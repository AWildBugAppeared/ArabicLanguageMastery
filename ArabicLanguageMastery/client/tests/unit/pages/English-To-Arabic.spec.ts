/* eslint-disable  @typescript-eslint/no-explicit-any */

// Imports
import Vue from 'vue';
import Vuetify from 'vuetify';

// Utilities
import { createLocalVue, mount, Wrapper } from '@vue/test-utils';

import EnglishToArabic from '@/pages/English-To-Arabic.vue';

describe('English-To-Arabic.vue', () => {
  const localVue = createLocalVue();
  let wrapper: Wrapper<Vue & { [key: string]: any }>;
  let vuetify: Vuetify;

  const mountWrapper = () => {
    wrapper = mount(EnglishToArabic, {
      localVue,
      vuetify,
      data() {
        return {};
      },
      mocks: {},
    });
  };

  beforeEach(() => {
    vuetify = new Vuetify({
      mocks: {
        $vuetify: {},
      },
    });
    Vue.use(Vuetify);
    document.body.setAttribute('data-app', 'true');
  });

  afterEach(() => {
    wrapper.destroy();
    jest.resetModules();
    jest.clearAllMocks();
  });

  it('should convert phonetic English to Arabic', async () => {
    mountWrapper();
    await wrapper.vm.$nextTick();

    const textArea = wrapper.find('[data-phonetic-text-area]');
    textArea.setValue('wa ##Allah##u Aakbaru');
    await wrapper.vm.$nextTick();

    const arabicText = wrapper.find('[data-arabic]');

    expect(arabicText.html()).toMatchSnapshot();
  });
});
