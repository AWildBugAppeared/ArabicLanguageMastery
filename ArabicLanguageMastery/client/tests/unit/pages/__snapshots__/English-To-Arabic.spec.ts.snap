// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`English-To-Arabic.vue should convert phonetic English to Arabic 1`] = `"<div dir=\\"rtl\\" data-arabic=\\"\\" class=\\"arabic converted-text pa-0 ma-0\\">وَ ﷲُ أَكْبَرُ</div>"`;
